source/MaterialXShaderGen/Shader.cpp

```diff
@@ -68,12 +68,9 @@ void Shader::initialize(ElementPtr element, ShaderGenerator& shadergen, const Sg
             {
                 if (!input->connection)
                 {
-                    // Check if the data type has a valid default value,
-                    // otherwise we can't assign user values to it anyway. 
-                    // This is the case for BSDF, EDF, VDF, and shader types 
-                    // which we never want to publish as editable uniforms.
-                    const string& typeDefault = shadergen.getSyntax()->getDefaultValue(input->type, true);
-                    if (!typeDefault.empty())
+                    // Check if the type is editable otherwise we can't 
+                    // publish the input as an editable uniform.
+                    if (input->type->isEditable())
```

Comment:
Make this test more robust by checking the TypeDesc if the type is editable. Examples of non-editable types are BSDF, EDF, surfaceshader, etc., we should never create uniforms for these.